lib/web_ui/dev/screenshot_manager.dart

```diff
@@ -0,0 +1,224 @@
+    }
+    // Temporary directories are deleted in the clenaup phase of after `felt`
+    // runs the tests.
+    temporaryDirectories.add(environment.webUiSimulatorScreenshotsDirectory);
+  }
+
+  /// Height of the part to crop from the top of the image.
+  ///
+  /// `xcrun simctl` command takes the screenshot of the entire simulator. We
+  /// are cropping top bit from screenshot, otherwise due to the clock on top of
+  /// the screen, the screenshot will differ between each run.
+  /// Note that this gap can change per phone and per iOS version. For more
+  /// details refer to `browser_lock.yaml` file.
+  int _heightOfHeader;
+
+  /// Height of the part to crop from the bottom of the image.
+  ///
+  /// This area is the footer navigation bar of the phone, it is not the area
+  /// used by tests (which is inside the browser).
+  int _heightOfFooter;
+
+  /// Used as a suffix for the temporary file names used for screenshots.
+  int _fileNameCounter = 0;
+
+  /// Capture a screenshot of entire simulator.
+  ///
+  /// Example screenshot with dimensions: W x H.
+  ///
+  ///  <----------  W ------------->
+  ///  _____________________________
+  /// | Phone Top bar (clock etc.)  |   Ʌ
+  /// |_____________________________|   |
+  /// | Broswer search bar          |   |
+  /// |_____________________________|   |
+  /// | Web page content            |   |
+  /// |                             |   |
+  /// |                             |
+  /// |                             |   H
+  /// |                             |
+  /// |                             |   |
+  /// |                             |   |
+  /// |                             |   |
+  /// |                             |   |
+  /// |_____________________________|   |
+  /// | Phone footer bar            |   |
+  /// |_____________________________|   V
+  ///
+  /// After taking the screenshot top and bottom parts of the image by
+  /// [_heightOfHeader] and [_heightOfFooter] consecutively. Hence web content
+  /// has the dimensions:
+  ///
+  /// W x (H - [_heightOfHeader] - [_heightOfFooter])
+  ///
+  /// [region] is used to decide which part of the web content will be used in
+  /// test image. It includes starting coordinate x,y as well as height and
+  /// width of the area to capture.
+  ///
+  /// Uses simulator tool `xcrun simctl`'s 'screenshot' command.
+  Future<Image> capture(Rectangle region) async {
+    final String filename = 'screenshot${_fileNameCounter}.png';
+    _fileNameCounter++;
+
+    await IosSafariArgParser.instance.iosSimulator.takeScreenshot(
+        filename, environment.webUiSimulatorScreenshotsDirectory);
+
+    final io.File file = io.File(path.join(
+        environment.webUiSimulatorScreenshotsDirectory.path, filename));
+    List<int> imageBytes;
+    if (!file.existsSync()) {
+      throw Exception('Failed to read the screenshot '
+          'screenshot${_fileNameCounter}.png.');
+    }
+    imageBytes = await file.readAsBytes();
```

Comment:
Do we need the file after reading the image bytes? If not, we should probably delete it right after this line.

Reply:
The files are all deleted in the cleanup time.

Reply:
done